kernel/network/routing.c

```diff
@@ -0,0 +1,92 @@
+
+/******************************************************************************
+ *       ninjastorms - shuriken operating system                              *
+ *                                                                            *
+ *    Copyright (C) 2013 - 2016  Andreas Grapentin et al.                     *
+ *                                                                            *
+ *    This program is free software: you can redistribute it and/or modify    *
+ *    it under the terms of the GNU General Public License as published by    *
+ *    the Free Software Foundation, either version 3 of the License, or       *
+ *    (at your option) any later version.                                     *
+ *                                                                            *
+ *    This program is distributed in the hope that it will be useful,         *
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of          *
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the           *
+ *    GNU General Public License for more details.                            *
+ *                                                                            *
+ *    You should have received a copy of the GNU General Public License       *
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>.   *
+ ******************************************************************************/
+
+#include "kernel/logger/logger.h"
+#include "kernel/network/ethernet.h"
+#include "kernel/network/routing.h"
+#include "kernel/network/arp.h"
+#include <stdlib.h>
+
+uint32_t next_slot;
+arp_table_entry_t *arp_table;
+
+mac_address_t
+get_mac_for_ip(uint32_t ip, uint8_t send_request)
```

Comment:
I'm unsure about having that `send_request` Boolean there. I think the usage will always be like call once with true and then in while 1 loop with false. Feels like this is unintuitive and this method does more than it should.

Reply:
We could split it in two functions: one that will look in the arp table and send an arp request if the ip is not found and another one that only looks in the arp table (so that we don't flood the network with arp requests)

Reply:
I splitted the functionality in the functions `arp_get_mac` and `arp_table_lookup`. Please have a look :)